libcloud/backup/base.py

```diff
@@ -0,0 +1,488 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from libcloud.common.base import ConnectionUserAndKey, BaseDriver
+from libcloud.backup.types import BackupTargetType
+
+__all__ = [
+    'BackupTarget',
+    'BackupDriver',
+    'BackupTargetJob',
+    'BackupTargetRecoveryPoint'
+]
+
+
+class BackupTarget(object):
+    """
+    A backup target
+    """
+
+    def __init__(self, id, name, address, type, driver, extra=None):
+        """
+        :param id: Target id
+        :type id: ``str``
+
+        :param name: Name of the target
+        :type name: ``str``
+
+        :param address: Hostname, FQDN, IP, file path etc.
```

Comment:
Hm, trying to think of a better name for this argument. Maybe `value` or `target` (I see how `target` could be confusing though).